### DISCUSSION TITLE
Pilot to wait for a selector to be visible

### DISCUSSION THREAD

**Discussion by SebastienSyd** [Textualize/textual#2351] [Q&A]
Hello,

I am using the pilot to validate my Textual app is working correctly with key presses and click. It works wonderfully.
1 thing which is not perfect is that I need to use the `pause()` which lengthens the duration of the tests the more I add. Is there a simple way to basically `wait for a selector to be visible`? That would spare several seconds per test, and avoid false positives when the pause requires - sometimes - to be longer that the static value I set

thank you 🙏

**Comment by davep** [comment]
Could you give a minimal stand-alone working example of the sort of test you're writing? It would be easier to offer suggestions and advice if we can see what you're working with.
